code/xcvm/lib/proto/src/lib.rs

```diff
@@ -0,0 +1,452 @@
+			assets: Funds(
+				transfer
+					.assets
+					.into_iter()
+					.map(|asset| asset.try_into())
+					.collect::<core::result::Result<Vec<_>, _>>()?,
+			),
+		})
+	}
+}
+
+impl<Acc> From<transfer::AccountType> for Destination<Acc>
+where
+	Acc: From<Vec<u8>>,
+{
+	fn from(account_type: transfer::AccountType) -> Self {
+		match account_type {
+			transfer::AccountType::Account(Account { account }) =>
+				Destination::Account(account.into()),
+			transfer::AccountType::Relayer(_) => Destination::Relayer,
+		}
+	}
+}
+
+impl TryFrom<Asset> for (xcvm_core::AssetId, Amount) {
+	type Error = ();
+
+	fn try_from(asset: Asset) -> core::result::Result<Self, Self::Error> {
+		let asset_id = asset.asset_id.ok_or(())?.try_into()?;
+		let amount = asset.balance.ok_or(())?.try_into()?;
+
+		Ok((asset_id, amount))
+	}
+}
+
+impl TryFrom<AssetId> for xcvm_core::AssetId {
+	type Error = ();
+
+	fn try_from(asset_id: AssetId) -> core::result::Result<Self, Self::Error> {
+		Ok(xcvm_core::AssetId(asset_id.asset_id.ok_or(())?.into()))
+	}
+}
+
+impl TryFrom<Balance> for Amount {
+	type Error = ();
+
+	fn try_from(balance: Balance) -> core::result::Result<Self, Self::Error> {
+		let balance_type = balance.balance_type.ok_or(())?;
+		let wrap = |num: u128| -> FixedU128<U16> { FixedU128::wrapping_from_num(num) };
+		// TODO(aeryz): This can be a helper function in SDK so that users won't
+		// necesarilly need to know how the ratio is handled in our SDK
+		// Calculates `x` in the following equation: nom / denom = x / max
+		let calc_nom = |nom: u128, denom: u128, max: u128| -> u128 {
+			wrap(nom)
+				.saturating_div(wrap(denom))
+				.saturating_mul(wrap(max))
+				.wrapping_to_num::<u128>()
+		};
+		match balance_type {
+			balance::BalanceType::Ratio(Ratio { nominator, denominator }) => {
+				let nominator = nominator.ok_or(())?;
+				let denominator = denominator.ok_or(())?;
+				Ok(Amount::ratio(calc_nom(nominator.into(), denominator.into(), MAX_PARTS)))
+			},
+			balance::BalanceType::Absolute(Absolute { value }) => {
+				let value = value.ok_or(())?;
+				Ok(Amount::absolute(value.into()))
+			},
+			balance::BalanceType::Unit(Unit { integer, ratio }) => {
+				let integer = integer.ok_or(())?;
+				let ratio = ratio.ok_or(())?;
+				Ok(Amount::new(
+					integer.into(),
+					calc_nom(
+						ratio.nominator.ok_or(())?.into(),
+						ratio.denominator.ok_or(())?.into(),
+						MAX_PARTS,
+					),
+				))
+			},
+		}
+	}
+}
+
+// XCVM types to Protobuf conversion
+
+impl From<Amount> for Balance {
+	fn from(amount: Amount) -> Self {
+		// Note that although functionally nothing changes, there is no guarantee of getting the
+		// same protobuf when you convert protobuf to XCVM types and convert back again. Because
+		// `intercept = 0 & ratio = 0` is always converted to `Absolute`. But this can be also
+		// expressed with `Ratio` and `Unit` as well. Also, since the ratio is expanded to use
+		// denomitor `MAX_PARTS`, it also won't be the same.
```

Comment:
```suggestion
		// denominator `MAX_PARTS`, it also won't be the same.
```
Nit spelling, failing spellcheck CI